File: Chapter 14/JS/section_14_6_instrumentation_logging/tests/orderLambdaStubTests.js
/**
 * Listing 14.3 — Pattern A: The Minimal Hand-Coded Lambda Stub
 *
 * Book file: com/ecommerce/order/tests/OrderLambdaStubTests.java
 */
import { test } from "node:test";
import assert from "node:assert";
import { HappyPathPaymentPort } from "../ports/happyPathPaymentPort.js";
import { OrderService } from "../services/orderService.js";

test("order checkout with inline stub", () => {
  // Create a sterile, static Test Double with zero network overhead
  const inlinePaymentStub = new HappyPathPaymentPort();
  const service = new OrderService(inlinePaymentStub);

  const result = service.checkout("ord_99812", 150.0);
  assert.strictEqual(result, true, "Checkout failed under a cooperative happy-path stub.");
});

File: Chapter 14/C#/Chapter14.Tests/OrderLambdaStubTests.cs
using Chapter14.Section_14_6_InstrumentationLogging.Ports;
using Chapter14.Section_14_6_InstrumentationLogging.Services;
using Xunit;

namespace Chapter14.Tests;

/// <summary>
/// Listing 14.3 — Pattern A: The Minimal Hand-Coded Stub
///
/// Book file: com/ecommerce/order/tests/OrderLambdaStubTests.java
///
/// NOTE: The book's Java example uses an inline lambda (Java interfaces with a
/// single method are functional). C# interfaces are NOT functional types, so we
/// use an equivalent hand-coded concrete stub class (HappyPathPaymentPort)
/// that encapsulates the exact same "always return true" behavior.
/// </summary>
public class OrderLambdaStubTests
{
    [Fact]
    public void TestOrderCheckoutWithInlineStub()
    {
        // Create a sterile, static Test Double with zero network overhead
        IPaymentPort inlinePaymentStub = new HappyPathPaymentPort();
        var service = new OrderService(inlinePaymentStub);

        bool result = service.Checkout("ord_99812", 150.00);
        Assert.True(result, "Checkout failed under a cooperative happy-path stub.");
    }
}
